From f4b552f272034bd06b213ab5a5e2bfec179af124 Mon Sep 17 00:00:00 2001
From: Sai Krishna Potthuri <lakshmi.sai.krishna.potthuri@xilinx.com>
Date: Tue, 28 Apr 2020 19:49:30 +0530
Subject: [PATCH 1355/1852] mtd: cadence-quadspi: Added support for GD25LX256E

commit 46fe0933d53d4e753ed8ba0570b358c988425617 from
https://github.com/Xilinx/linux-xlnx.git

This patch adds support for ospi Gigadevice part(GD25LX256E).

Signed-off-by: Sai Krishna Potthuri <lakshmi.sai.krishna.potthuri@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/mtd/spi-nor/spi-nor.c | 8 +++++++-
 include/linux/mtd/cfi.h       | 1 +
 include/linux/mtd/spi-nor.h   | 1 +
 3 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/drivers/mtd/spi-nor/spi-nor.c b/drivers/mtd/spi-nor/spi-nor.c
index b69457adf5cc..bc078d712916 100644
--- a/drivers/mtd/spi-nor/spi-nor.c
+++ b/drivers/mtd/spi-nor/spi-nor.c
@@ -2679,6 +2679,11 @@ static const struct flash_info spi_nor_ids[] = {
 			     SECT_4K | USE_FSR | SPI_NOR_OCTAL_READ |
 			     SPI_NOR_OCTAL_WRITE | SPI_NOR_4B_OPCODES) },
 
+	/* Cypress */
+	{ "GD25LX256E",  INFO(0xc86819, 0, 64 * 1024, 512,
+			     SECT_4K | USE_FSR | SPI_NOR_OCTAL_READ |
+			     SPI_NOR_OCTAL_WRITE | SPI_NOR_4B_OPCODES) },
+
 	/* PMC */
 	{ "pm25lv512",   INFO(0,        0, 32 * 1024,    2, SECT_4K_PMC) },
 	{ "pm25lv010",   INFO(0,        0, 32 * 1024,    4, SECT_4K_PMC) },
@@ -5523,7 +5528,8 @@ int spi_nor_scan(struct spi_nor *nor, const char *name,
 	if (info->flags & SPI_NOR_HAS_LOCK)
 		nor->flags |= SNOR_F_HAS_LOCK;
 
-	if ((u16)JEDEC_MFR(nor->info) != SNOR_MFR_MICRON)
+	if (((u16)JEDEC_MFR(nor->info) != SNOR_MFR_MICRON) &&
+	    ((u16)JEDEC_MFR(nor->info) != SNOR_MFR_CYPRESS))
 		nor->flags |= SNOR_F_BROKEN_OCTAL_DDR;
 
 	/*
diff --git a/include/linux/mtd/cfi.h b/include/linux/mtd/cfi.h
index 208c87cf2e3e..acbb7c18e4c9 100644
--- a/include/linux/mtd/cfi.h
+++ b/include/linux/mtd/cfi.h
@@ -366,6 +366,7 @@ struct cfi_fixup {
 #define CFI_MFR_MICRON		0x002C /* Micron */
 #define CFI_MFR_TOSHIBA		0x0098
 #define CFI_MFR_WINBOND		0x00DA
+#define CFI_MFR_CYPRESS		0x00C8
 
 void cfi_fixup(struct mtd_info *mtd, struct cfi_fixup* fixups);
 
diff --git a/include/linux/mtd/spi-nor.h b/include/linux/mtd/spi-nor.h
index f95d41bb8b2a..5e873c52c4f8 100644
--- a/include/linux/mtd/spi-nor.h
+++ b/include/linux/mtd/spi-nor.h
@@ -27,6 +27,7 @@
 #define SNOR_MFR_SST		CFI_MFR_SST
 #define SNOR_MFR_WINBOND	0xef /* Also used by some Spansion */
 #define SNOR_MFR_ISSI		CFI_MFR_PMC
+#define SNOR_MFR_CYPRESS	CFI_MFR_CYPRESS
 
 /*
  * Note on opcode nomenclature: some opcodes have a format like
-- 
2.31.1

